sympy/ntheory/generate.py

```diff
@@ -137,6 +139,84 @@ def primerange(self, a, b):
             else:
                 return
 
+    def totientrange(self, a, b):
+        """Generate all totient numbers for the range [a, b).
+
+        Examples
+        ========
+
+        >>> from sympy import sieve
+        >>> print([i for i in sieve.totientrange(7, 18)])
+        [6, 4, 6, 4, 10, 4, 12, 6, 8, 8, 16]
+        """
+        from sympy.functions.elementary.integers import ceiling
+
+        # wrapping ceiling in int will raise an error if there was a problem
+        # determining whether the expression was exactly an integer or not
+        a = max(1, int(ceiling(a)))
```

Comment:
perhaps `as_int(ceiling(a))` would be better since it will throw a meaningful error (albeit only slightly more helpful)?

Reply:
Okay, so `int(ceiling(a))` might need change throughout the file.